docs/workload-development-kit/fabric-data-plane.md

```diff
@@ -12,41 +12,98 @@ ms.date: 05/21/2024
+1. Acquire a token with SQL scopes on behalf of the user. Again, in order to successfully establish a SQLConnection, you must use the scope https://database.windows.net//user_impersonation.
+1. Use the SQL token and connection information to open a SQL Connection:
+
+    ```csharp
+    private SqlConnection GetSqlConnection(string token, string databaseFqdn, string initialCatalog)
+            {
+                var builder = new SqlConnectionStringBuilder();
+                builder.DataSource = databaseFqdn; // ConnectionInfo from Get Warehouse API
+                builder.InitialCatalog = initialCatalog; //display name of the Warehouse
+                builder.ConnectTimeout = 120;
+    
+                var connection = new SqlConnection();
+                connection.AccessToken = token; // SQL token acquired with the Azure SQL Database user_impersonation scope
+                connection.ConnectionString = builder.ConnectionString;
+                connection.Open();
+                return connection;
+            }
+    ```
+
+1. This connection can now be queried in order to access data from the Warehouse. For more information on utilizing the Microsoft.Data.SqlClient namespace, reference the [Microsoft.Data.SqlClient Namespace Documentation](/dotnet/api/microsoft.data.sqlclient).
+
 ### Writing data 
 
 In addition to reading data using the token, you can also use ADLS APIs to write data into tables as described by the [Delta Lake protocol](https://github.com/delta-io/delta/blob/master/PROTOCOL.md).
 
 You can also use the APIs to create files and directories.
 
-Alternatively, you can use other Fabric workloads to write data to the platform. For example, you can use Fabric's Lakehouse workload API to efficiently [load common file types to an optimized Delta table](../data-engineering/load-to-tables.md). This is done by sending a [POST or PUT request](/rest/api/fabric/lakehouse/tables/load-table?tabs=HTTP) to the appropriate API endpoint. For detailed information on how to load data to table using the Lakehouse, you can refer to the Fabric API reference. 
-
-### Read Metadata
-
-Fabric REST APIs also provide you with a way to access the different properties of items. For example, querying the [Get Lakehouse API](/rest/api/fabric/lakehouse/items/get-lakehouse?tabs=HTTP) provides you with the metadata for a certain Lakehouse, including useful properties such as OneLake paths and the SQL connection string.
+Alternatively, you can use other Fabric workloads to write data to the platform. For example, you can use Fabric's Lakehouse workload API to efficiently [load common file types to an optimized Delta table](../data-engineering/load-to-tables.md). This is done by sending a [POST or PUT request](/rest/api/fabric/lakehouse/tables/load-table?tabs=HTTP) to the appropriate API endpoint. For detailed information on how to load data to table using the Lakehouse, you can refer to the Fabric API reference.
```

Comment:
Here and elsewhere - remove '?tabs=HTTP' from URLs.

'appropriate' -- > remove.

This sentence isn't clear - 'For detailed information on how to load data to table using the Lakehouse, you can refer to the Fabric API reference.'

If you want to send the customer somewhere, add a link.